### ISSUE TITLE
PyPDF2 decrypt does not work for PDFs with spaces in the file name

### ISSUE THREAD

**Issue by lee12jin** [py-pdf/PyPDF2#144]
http://stackoverflow.com/questions/26242952/pypdf-2-decrypt-not-working/26243460#26243460

I already know why PyPDF2 decrypt won't work for some PDFs.
Basically if the PDF file name has a space (or in regex terms "\s"),
then decryption seems to fail somewhere.

Decryption would return successful (return an int 1). But, the logger would still
raise an error saying "File has not been decrypted".

Basically, this is just a heads up for others. And maybe for other fellow developers
to change the code to handle this. :+1: :)

**Comment by louib** [comment]
Tested with 1.25.1.
I'm not sure this issue is caused by the file name. The real problem seems to be that decrypt() does not allow empty passwords while encrypt() does. I tested with file names containing special characters, and encryption / decryption seems to work fine, as long as the password is not empty.

**Comment by MartinThoma** [comment]
This is not / no longer the case with `PyPDF2==2.3.1`

```
>>> from PyPDF2 import PdfReader
>>> reader = PdfReader("crazyones password.pdf", password="password"); print(len(reader.pages))
1
>>> 
>>> reader = PdfReader("crazyones protected-ilovepdf.pdf", password="password"); print(len(reader.pages))
1
>>> reader = PdfReader("r6 both-passwords.pdf", password="password"); print(len(reader.pages))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_reader.py", line 272, in __init__
    raise PdfReadError("Wrong password")
PyPDF2.errors.PdfReadError: Wrong password
>>> reader = PdfReader("r6 both-passwords.pdf", password="foo"); print(len(reader.pages))
1
>>> reader = PdfReader("r6 both-passwords.pdf", password="bar"); print(len(reader.pages))
1
```
